06a-apply_ica.py

```diff
@@ -0,0 +1,179 @@
+from mne.preprocessing import create_eog_epochs, create_ecg_epochs
+from mne.report import Report
+
+import numpy as np
+import config
+
+
+def run_evoked(subject):
+    print("Processing subject: %s" % subject)
+    meg_subject_dir = op.join(config.meg_dir, subject)
+
+    # load epochs to reject ICA components
+    extension = '-epo'
+    fname_in = op.join(meg_subject_dir,
+                       config.base_fname.format(**locals()))
+    epochs = mne.read_epochs(fname_in, preload=True)
+
+    extension = 'cleaned-epo'
+    fname_out = op.join(meg_subject_dir,
+                        config.base_fname.format(**locals()))
+
+    print("Input: ", fname_in)
+    print("Output: ", fname_out)
+
+    # load first run of raw data for ecg /eog epochs
+    raw_list = list()
+    print("  Loading one run from raw data")
+    extension = config.runs[0] + '_sss_raw'
+    raw_fname_in = op.join(meg_subject_dir,
+                           config.base_fname.format(**locals()))
+    raw = mne.io.read_raw_fif(raw_fname_in, preload=True)
+
+    # run ICA on MEG and EEG
+    picks_meg = mne.pick_types(raw.info, meg=True, eeg=False,
+                               eog=False, stim=False, exclude='bads')
+    picks_eeg = mne.pick_types(raw.info, meg=False, eeg=True,
+                               eog=False, stim=False, exclude='bads')
+    all_picks = {'meg': picks_meg, 'eeg': picks_eeg}
+
+    for ch_type in ['meg', 'eeg']:
+        print(ch_type)
+        picks = all_picks[ch_type]
+
+        # Load ICA
+        fname_ica = op.join(meg_subject_dir,
+                            '{0}_{1}_{2}-ica.fif'.format(subject, config.study_name,
+                                                         ch_type))
+        print('Reading ICA: ' + fname_ica)
+        ica = read_ica(fname=fname_ica)
+
+        pick_ecg = mne.pick_types(raw.info, meg=False, eeg=False,
+                                  ecg=True, eog=False)
+
+        # ECG
+        # either needs an ecg channel, or avg of the mags (i.e. MEG data)
+        if pick_ecg or ch_type == 'meg':
+
+            picks_ecg = np.concatenate([picks, pick_ecg])
+
+            # Create ecg epochs
+            if ch_type == 'meg':
+                reject = {'mag': config.reject['mag'],
+                          'grad': config.reject['grad']}
+            elif ch_type == 'eeg':
+                reject = {'eeg': config.reject['eeg']}
+
+            ecg_epochs = create_ecg_epochs(raw, picks=picks_ecg, reject=reject,
+                                           baseline=(None, 0), tmin=-0.5,
+                                           tmax=0.5)
+
+            ecg_average = ecg_epochs.average()
+
+            # XXX I had to lower the threshold for ctps (default 0.25), otherwise it does not
+            # find any components
+            # check how this behaves on other data
+            ecg_inds, scores = ica.find_bads_ecg(ecg_epochs, method='ctps',
+                                                 threshold=0.1)
```

Comment:
@agramfort 
I had to reduce the threshold for ctps, otherwise to rather obviously outstanding components did not get selected.